### PR TITLE
Bug #202176, Bug #202150, Task #202097, Task #201707, #202125

### PR DESCRIPTION
**Pull request by sagarkoshti1990**
- Bug #202176 - https://tracker.tekdi.net/issues/202176
- Bug #202150 - https://tracker.tekdi.net/issues/202150
- Bug #202125 - https://tracker.tekdi.net/issues/202125

- Task #202097 - https://tracker.tekdi.net/issues/202097
- Task #201707 - https://tracker.tekdi.net/issues/201707

### CODE REVIEW

File: apps/front-end/src/component/BaseInput.js [hunk 1]
```diff
@@ -652,11 +656,99 @@ const templates = {
...
+        let title = schemaItem.label ? schemaItem.label : schemaItem.title;
+        if (schemaItem?.format === "FileUpload") {
+          error.message = `${t("REQUIRED_MESSAGE_UPLOAD")} "${t(title)}"`;
+        } else {
+          error.message = `${t("REQUIRED_MESSAGE")} "${t(title)}"`;
+        }
+      } else {
+        error.message = `${t("REQUIRED_MESSAGE")}`;
+      }
+    } else if (error.name === "minItems") {
```
Comment (sagarkoshti1990) [inline apps/front-end/src/component/BaseInput.js:697]: use sprintf("कमसे कम %s %s चुनें", "1", "विषय")
sprintf("Choose %s %s minimum", "1", "subject")

File: apps/front-end/src/component/BaseInput.js [hunk 2]
```diff
@@ -652,11 +656,99 @@ const templates = {
...
+      }
+    }
+  }
+};
+
+export const focusToField = (errors) => {
+  const firstError = errors?.[0];
+  scrollToField(firstError);
+};
+
```
Comment (sagarkoshti1990) [inline apps/front-end/src/component/BaseInput.js:682]: add comments

File: apps/front-end/src/pages/front-end/BenificiaryEnrollment.js [hunk 3]
```diff
@@ -34,224 +35,184 @@ export default function BenificiaryEnrollment() {
...
-          borderRadius="10px"
-          borderWidth="1px"
-          bg="white"
-          borderColor="appliedColor"
-        >
-          <HStack justifyContent="space-between" alignItems="Center">
-            <FrontEndTypo.H3 bold color="textGreyColor.800">
-              {t("UPLOAD_RECEIPT")}
-            </FrontEndTypo.H3>
+const ItemComponent = ({ title, item, onlyField, onEdit, step, notShow }) => {
```
Comment (manojLondhe) [inline apps/front-end/src/pages/front-end/BenificiaryEnrollment.js:143]: move to common location

Comment (sagarkoshti1990) [inline apps/front-end/src/pages/front-end/BenificiaryEnrollment.js:143]: move this to conman component.

File: apps/front-end/src/pages/front-end/Form.js [hunk 4]
```diff
@@ -336,73 +282,81 @@ export default function App({ facilitator, ip, onClick }) {
...
-        errors?.dob?.addError(t("MINIMUM_AGE_18_YEAR_OLD"));
-      }
-    }
-    ["grampanchayat", "first_name", "last_name", "middle_name"].forEach(
-      (key) => {
+  const validate = (data, key) => {
+    let error = {};
+    switch (key) {
+      case "mobile":
+        if (data?.mobile?.toString()?.length !== 10) {
```
Comment (manojLondhe) [inline apps/front-end/src/pages/front-end/Form.js:289]: use form lib correctly, add task to tracker backlog

**Review by manojLondhe — CHANGES_REQUESTED** [review]

File: apps/front-end/src/pages/front-end/Form.js (hunk 4 above)
Comment (sagarkoshti1990) [reply apps/front-end/src/pages/front-end/Form.js:289]: varun is adding.

**Pull request merged**